pkg/broker/config/config.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"github.com/golang/protobuf/proto"
+)
+
+// ReadOnlyTargets provides "read" functions for broker targets.
+type ReadOnlyTargets interface {
+	// RangeNamespace ranges over targets in the given namespace.
+	RangeNamespace(namespace string, f func(t Target) bool)
+	// Range ranges over all targets.
+	Range(f func(t Target) bool)
+	// Bytes serializes all the targets.
+	Bytes() ([]byte, error)
+	// String returns the text format of all the targets.
+	String() string
+	// EqualsBytes checks if the current targets config equals the given
+	// targets config in bytes.
+	EqualsBytes([]byte) bool
+	// EqualsString checks if the current targets config equals the given
+	// targets config in string.
+	EqualsString(string) bool
+}
+
+// Targets provides "read" and "write" functions for broker targets.
+type Targets interface {
+	ReadOnlyTargets
+
+	// Union adds the given targets.
+	Union(...Target) Targets
+	// Except removes the give targets.
```

Comment:
This (and Union) should document whether it returns a new set or modifies the set in place. I think it's the latter, looking at the code.

Also, check out https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets for an existing set interface. It might be a good idea to be consistent. In the generated k8s sets, `Insert` adds items and `Delete` removes items. `Union` is a method for merging two sets.